pupil_src/shared_modules/video_capture/ndsi_backend.py

```diff
@@ -486,29 +489,34 @@ def auto_select_manager(self):
         super().auto_select_manager()
         self.notify_all(
             {
-                "subject": "notify.selected_host",
+                "subject": "backend.ndsi_host_selected",
                 "selected_host": self.selected_host,
                 "delay": 0.4,
             }
         )
 
-    def auto_activate(self):
-        cam_selection_lut = {"eye0": "ID0", "eye1": "ID1", "world": "ID2"}
-        cam_id = cam_selection_lut[self.g_pool.process]
+    def auto_activate_source(self):
+        if not self.selected_host:
+            return
 
         src_sel, src_sel_labels = self.source_selection_list()
-        if src_sel is None or src_sel_labels is None:
+        if len(src_sel) <= 1 and src_sel[0] is None:
+            logger.warning("No camera was found on the remote host.")
             return
 
-        source_id = [
-            src_sel[i] for i, lab in enumerate(src_sel_labels) if cam_id in lab
-        ]
-
-        if len(source_id) != 1:
-            return
+        cam_ids = self.cam_selection_lut[self.g_pool.process]
 
-        source_id = source_id[0]
-        self.activate(source_id)
+        for cam_id in cam_ids:
+            try:
+                source_id = next(
+                    src_sel[i] for i, lab in enumerate(src_sel_labels) if cam_id in lab
+                )
+            except StopIteration:
+                source_id = None
+                logger.warning("No camera was found with ID: {}".format(cam_id))
```

Comment:
In accordance with the button label change, I would suggest to rename the warning to `No default device found for {process name}`.

Additionally, I would move the error handling/warning out of the exception into the [`else`-part of the for-loop](http://book.pythontips.com/en/latest/for_-_else.html).

Reply:
Good suggestions. Thanks!